src/__experimental__/components/decimal/decimal.stories.js

```diff
@@ -38,47 +39,52 @@ const previous = {
   allowEmptyValue: false
 };
 
-function makeStory(name, themeSelector) {
-  const component = () => {
-    const precisionRange = {
-      range: true,
-      min: 0,
-      max: 15,
-      step: 1
-    };
-    const align = select(
-      'align',
-      OptionsHelper.alignBinary,
-      Decimal.defaultProps.align
-    );
-    const precision = number('precision', Decimal.defaultProps.precision, precisionRange);
-    const allowEmptyValue = boolean('allowEmptyValue', false);
-
-    // When the allowEmptyValue knob changes we want to force the component to re-create
-    // allowEmptyValue is only used in the constructor and it is not currently supported to change during the lifetime
-    // of the component
-    if (previous.allowEmptyValue !== allowEmptyValue) {
-      previous.key = guid();
-    }
-    previous.allowEmptyValue = allowEmptyValue;
-    const { key } = previous;
```

Comment:
Is it OK to remove this code which @jamime added in #2540 ?

Reply:
well spotted @kyanmichael

Reply:
Thanks for restoring that code :white_check_mark: